### DISCUSSION TITLE
Native image backwards compatibility

### DISCUSSION THREAD

**Discussion by lmb** [microsoft/ebpf-for-windows#3839] [Q&A]
I recently tried to load a .sys file compiled against roughly v0.18 against current main. This failed due to a mismatch of the hash of the program info. It seems like upgrading the runtime broke the old native image.

Was this breakage intentional? What is the plan for backwards compatibility in the future? Having to always load the .sys against the correct version of the runtime would be very limiting.

**Comment by shpalani** [comment]
Please reply to #3863

**Comment by shankarseal** [comment]
Let us use the issue #3863 for this discussion.
